velox/common/caching/CachedFactory.h

```diff
@@ -39,178 +40,418 @@
 
 namespace facebook::velox {
 
-// CachedFactory provides a thread-safe way of backing a keyed generator
-// (e.g. the key is filename, and the value is the file data) by a cache.
-//
-// Generator should take a single Key argument and return a Value;
-// The Value should be either a value type or should manage its own lifecycle
-// (shared_ptr). If it is not thread-safe it must do its own internal locking.
-template <typename Key, typename Value, typename Generator>
+/// A smart pointer that represents data that may be in a cache and is thus not
+/// owned, or is owned like a unique_ptr. We could also implement this by a
+/// unique_ptr with a custom deleter.
```

Comment:
/// A smart pointer that represents a cache entry. It can be owned like a unique_ptr.

Reply:
Remove the 'We could also implement this by a with a custom deleter.' part